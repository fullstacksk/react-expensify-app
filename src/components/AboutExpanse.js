import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Alert from 'react-bootstrap/Alert';

const AboutExpanse = () => (
    <Container>
        <Container>
            <Alert variant="success">
                <Row className="px-2">
                    <h5>About Expanse Manager</h5>
                    <p>Visual Studio Code and WebStorm support debugging out of the box with Create React App. This enables you as a developer to write and debug your React code without leaving the editor, and most importantly it enables you to have a continuous development workflow, where context switching is minimal, as you donâ€™t have to switch between tools.</p>
                </Row>
            </Alert>
        </Container>
    </Container>
);

export default AboutExpanse;