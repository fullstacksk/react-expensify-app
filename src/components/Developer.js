import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Alert from 'react-bootstrap/Alert';

const Developer = () => (
    <Container>
        <Container>
            <Alert variant="success" >
                <Row className="px-2">
                    <Col>
                        <h5>Fullstack Developer</h5>
                        <p>Hey, I am <code>SK Sahil</code> and this was the my first <kbd>React</kbd> web app.</p>
                        <h6>About React Js</h6>
                        <p>React Js is an wonderful js library <code>not a framework</code> which is  being used for dynamic front end development. It is developed by <code>Facebook</code> itself. It uses nodejs internally as programming language. NodeJs works on chromes <kbd>V8 Engine</kbd></p>
                    </Col>
                </Row>
            </Alert>
            <Alert variant="info" >
                <Row className="px-2">
                    <h5>Connect with me</h5>
                </Row>
                <Row>
                    <Col md={4}>
                        <a className="btn btn-primary" href="https://www.facebook.com/profile.php?id=100008404923891">Facebook</a>
                    </Col>
                    <Col md={4}>
                        <a className="btn btn-warning" href="https://www.instagram.com/sk_sahil_official/" >Instagram</a>
                    </Col>
                    <Col md={4}>
                        <a className="btn btn-secondary" href="https://github.com/fullstacksk/react-expensify-app">Github</a>
                    </Col>
                </Row>
            </Alert>
            <Alert variant="info">
                <Row>
                    <Col>
                        <h6>Suggest Me At</h6>
                        <kbd>fullstacksk@gmail.com</kbd></Col>
                </Row>
            </Alert>
        </Container>
    </Container >
);

export default Developer;