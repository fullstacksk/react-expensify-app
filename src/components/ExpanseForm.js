import React from 'react';
import moment from 'moment';
import Form from 'react-bootstrap/Form';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Alert from 'react-bootstrap/Alert';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/initialize';

export default class ExpanseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: props.expanse ? props.expanse.id : '',
            description: props.expanse ? props.expanse.description : "",
            note: props.expanse ? props.expanse.note : "",
            amount: props.expanse ? props.expanse.amount.toString() : "",
            createdAt: props.expanse ? moment(props.expanse.craetedAt) : moment(),
            calenderFocused: false,
            error: ''
        }
    }
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: "Please provide description and amount!!" }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                description: this.state.description,
                note: this.state.note,
                amount: parseFloat(this.state.amount),
                createdAt: this.state.createdAt.valueOf()
            })
        }
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    }
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    }
    onAmountChange = (e) => {

        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/))
            this.setState(() => ({ amount }))
    }
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calenderFocused: focused }));
    }
    render() {
        return (
            <Container>
                <Row className="justify-content-md-center">
                    <Col md={6}>
                        {
                            this.state.error &&
                            <Alert variant="warning">
                                {this.state.error}
                            </Alert>
                        }
                        <Form onSubmit={this.onSubmit} >
                            <Form.Group >
                                <Form.Label>Description</Form.Label>
                                <Form.Control
                                    type="text"
                                    id="description"
                                    value={this.state.description}
                                    onChange={this.onDescriptionChange}
                                />
                            </Form.Group>
                            <Form.Group >
                                <Form.Label>Note</Form.Label>
                                <Form.Control
                                    as="textarea"
                                    rows="2"
                                    id="note"
                                    value={this.state.note}
                                    onChange={this.onNoteChange}
                                />
                            </Form.Group>
                            <Form.Group as={Row}>
                                <Col>
                                    <Form.Label>Amount</Form.Label>
                                    <Form.Control
                                        type="text"
                                        id="amount"
                                        value={this.state.amount}
                                        onChange={this.onAmountChange}
                                    />
                                </Col>
                                <Col id="singleDatePicker">
                                    <Form.Label>Created At</Form.Label>
                                    <br />
                                    <SingleDatePicker
                                        date={this.state.createdAt}
                                        onDateChange={this.onDateChange}
                                        focused={this.state.calenderFocused}
                                        onFocusChange={this.onFocusChange}
                                        numberOfMonths={1}
                                        isOutsideRange={() => false}
                                        anchorDirection="right"
                                    />
                                </Col>
                            </Form.Group>
                            <Form.Group as={Row}>
                                <Col sm={6} className="mb-1">
                                    <Button variant="info" className="w-100" type="submit">Add Expanse </Button>
                                </Col>
                                {this.props.expanse &&
                                    <Col sm={6}>
                                        <Button
                                            id="removeExpanse"
                                            variant="success"
                                            className="w-100"
                                            onClick={this.props.startRemoveExpanse}
                                        >Remove Expanse </Button>
                                    </Col>}
                            </Form.Group>
                        </Form>
                    </Col>
                </Row>
            </Container>
        )
    }
}
