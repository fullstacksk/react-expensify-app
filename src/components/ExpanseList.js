import React from 'react';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import slectExpanses from '../slectors/expanses';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Badge from 'react-bootstrap/Badge';
import Alert from 'react-bootstrap/Alert';

export const ExpanseList = (props) => (
    <Container>
        <Container>
            <Alert variant="warning" className="mb-1 py-2">
                <Row className="justify-content-between px-2">
                    <div><h5>Expanse<Badge variant="success" className="ml-2">{props.expanses.length}</Badge></h5></div>
                    <div><h5>Amount<Badge variant="warning" className="ml-2">
                        {
                            props.expanses.length ?
                                props.expanses.map(
                                    ({ amount = '0' }) => parseFloat(amount)
                                ).reduce((a = 0, v = 0) => a + v)
                                : 0
                        }
                    </Badge></h5></div>
                </Row>
            </Alert>
        </Container>
        {
            props.expanses.length ?

                <Container>
                    {
                        props.expanses.map((expanse, idx) => (
                            <NavLink
                                to={`/edit/${expanse.id}`}
                                key={idx}
                            >

                                <Alert variant="info" className="mb-2 shadow-sm ">
                                    <Row className="justify-content-between px-2">
                                        <div>
                                            <h5 className="pb-0">{expanse.description}</h5>
                                            <span className="small mt-0">{moment(expanse.createdAt).format('Do MMM, YYYY')}</span>
                                        </div>
                                        <div><h5>{expanse.amount}</h5></div>
                                    </Row>
                                </Alert>
                            </NavLink>
                        ))
                    }
                </Container >
                :

                <Container>
                    <Alert variant="dark" className="mb-1 py-2">
                        <Row className="justify-content-center px-2">
                            <div><h6><Badge variant="success" className="ml-2">No</Badge> Expanses</h6></div>
                        </Row>
                    </Alert>
                </Container>
        }
    </Container >
);
const mapStateToProps = (state) => ({
    expanses: slectExpanses(state.expanses, state.filters),
    filters: state.filters
});
export default connect(mapStateToProps)(ExpanseList);