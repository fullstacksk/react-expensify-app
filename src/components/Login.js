import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import Card from 'react-bootstrap/Card';
import Accordion from 'react-bootstrap/Accordion';
import { connect } from 'react-redux';
import { startLogin, startFacebookLogin } from '../actions/auth';

export const Login = ({ startLogin, startFacebookLogin }) => (
    <Container fluid className="h-min-100 h-max-100 bg-light">
        <Row md={4} className="justify-content-center align-items-center h-min-100 h-max-100">
            <Form>
                <Button
                    id="loginWithGoogle"
                    variant="success"
                    type="button"
                    className="w-100"
                    onClick={startLogin}
                > Login With Google</Button>
                <p className="text-center my-2 p-1"><Badge variant="info">Or</Badge></p>
                <Button
                    id="loginWithFacebook"
                    variant="primary"
                    type="button"
                    className="w-100"
                    onClick={startFacebookLogin}
                > Login With Facebook</Button>
                <p className="text-center my-2 p-1"><Badge variant="info">Or</Badge></p>

                <Accordion >
                    <Card>
                        <Accordion.Toggle as={Card.Header} eventKey="0" className="p-0">
                            <Button variant="info" type="button" className="w-100" disabled> Login Anonymously</Button>
                        </Accordion.Toggle>
                        <Accordion.Collapse eventKey="0">
                            <Card.Body>
                                <Form.Group controlId="formBasicEmail">
                                    <Form.Label>Email address</Form.Label>
                                    <Form.Control type="email" placeholder="Enter email" />
                                    <Form.Text className="text-muted">We'll never share your email with anyone else.</Form.Text>
                                </Form.Group>

                                <Form.Group controlId="formBasicPassword">
                                    <Form.Label>Password</Form.Label>
                                    <Form.Control type="password" placeholder="Password" />
                                </Form.Group>
                                <Button variant="info" type="submit" className="w-100"> Submit</Button>
                            </Card.Body>
                        </Accordion.Collapse>
                    </Card>
                </Accordion>
            </Form>
        </Row>
    </Container>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);