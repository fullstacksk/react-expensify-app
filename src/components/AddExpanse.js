import React from 'react';
import ExpanseForm from './ExpanseForm';
import { startAddExpanse } from '../actions/expanses';
import { connect } from 'react-redux';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Alert from 'react-bootstrap/Alert';

export class AddExpanse extends React.Component {

    onSubmit = (expanse) => {
        this.props.startAddExpanse(expanse);
        this.props.history.push("/dashboard");
    };
    render() {
        return (
            <Container>
                <Container>
                    <Alert variant="info">
                        <Row className="text-success px-2">
                            <h5>You can add your expanse!</h5>
                        </Row>
                    </Alert>
                </Container>
                <ExpanseForm
                    onSubmit={this.onSubmit}
                />
            </Container>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    startAddExpanse: (expanse) => dispatch(startAddExpanse(expanse))
})

export default connect(undefined, mapDispatchToProps)(AddExpanse);