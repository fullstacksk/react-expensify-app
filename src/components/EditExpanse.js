import React from 'react';
import { connect } from 'react-redux';
import { startEditExpanse, startRemoveExpanse } from '../actions/expanses';
import ExpanseForm from './ExpanseForm';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Alert from 'react-bootstrap/Alert';

export class EditExpanse extends React.Component {

    onSubmit = (expanse) => {
        this.props.startEditExpanse(this.props.expanse.id, expanse);
        this.props.history.push('/dashboard');
    };
    startRemoveExpanse = () => {
        this.props.startRemoveExpanse(this.props.expanse.id);
        this.props.history.push('/dashboard');
    }
    render() {
        return (
            <Container>
                <Container className="pt-3 pb-1">
                    <Alert variant='info'>
                        <Row className="px-2">
                            <p className="text-success mb-0">There is a request to edit the expanse having id : <kbd className="bg-info">{this.props.expanse.id}</kbd></p>
                        </Row>
                    </Alert>
                </Container>
                <ExpanseForm

                    onSubmit={this.onSubmit}
                    expanse={this.props.expanse}
                    startRemoveExpanse={this.startRemoveExpanse}
                />
            </Container>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expanse: state.expanses.find((expanse) => expanse.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpanse: (id, expanse) => dispatch(startEditExpanse(id, expanse)),
    startRemoveExpanse: (id) => dispatch(startRemoveExpanse({ id }))
})
export default connect(mapStateToProps, mapDispatchToProps)(EditExpanse);