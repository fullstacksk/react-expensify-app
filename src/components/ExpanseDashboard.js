import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Alert from 'react-bootstrap/Alert';
import ExpanseList from './ExpanseList';

const ExpanseDashboard = () => (
    <div>
        <Container>
            <Container >
                <Alert variant="danger" >
                    <Row className="px-2">
                        <h5>Expanse Dashboard</h5>
                    </Row>
                </Alert>
            </Container>
        </Container>

        <ExpanseList />
    </div>
);

export default ExpanseDashboard;