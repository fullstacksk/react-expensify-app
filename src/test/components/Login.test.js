import React from 'react'
import { shallow } from 'enzyme'
import { Login } from '../../components/Login'

test("Should render Login correctly", () => {
    const wrapper = shallow(<Login startLogin={() => { }} />)
    expect(wrapper).toMatchSnapshot()
})

test("Should call startLogin", () => {
    const startLogin = jest.fn()
    const wrapper = shallow(<Login startLogin={startLogin} />)
    wrapper.find('#loginWithGoogle').simulate('click')
    expect(startLogin).toHaveBeenCalled()

})