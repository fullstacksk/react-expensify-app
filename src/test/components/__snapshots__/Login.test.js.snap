// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login correctly 1`] = `
<Container
  className="h-min-100 h-max-100 bg-light"
  fluid={true}
>
  <Row
    className="justify-content-center align-items-center h-min-100 h-max-100"
    md={4}
    noGutters={false}
  >
    <Form
      inline={false}
    >
      <Button
        active={false}
        className="w-100"
        disabled={false}
        id="loginWithGoogle"
        onClick={[Function]}
        type="button"
        variant="success"
      >
         Login With Google
      </Button>
      <p
        className="text-center my-2 p-1"
      >
        <Badge
          pill={false}
          variant="info"
        >
          Or
        </Badge>
      </p>
      <Button
        active={false}
        className="w-100"
        disabled={false}
        id="loginWithFacebook"
        type="button"
        variant="primary"
      >
         Login With Facebook
      </Button>
      <p
        className="text-center my-2 p-1"
      >
        <Badge
          pill={false}
          variant="info"
        >
          Or
        </Badge>
      </p>
      <ForwardRef>
        <Card
          body={false}
        >
          <ForwardRef
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "CardHeader",
                "render": [Function],
              }
            }
            className="p-0"
            eventKey="0"
          >
            <Button
              active={false}
              className="w-100"
              disabled={false}
              type="button"
              variant="info"
            >
               Login Anonymously
            </Button>
          </ForwardRef>
          <AccordionCollapse
            eventKey="0"
          >
            <CardBody>
              <FormGroup
                controlId="formBasicEmail"
              >
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Email address
                </FormLabel>
                <FormControl
                  placeholder="Enter email"
                  type="email"
                />
                <FormText
                  className="text-muted"
                >
                  We'll never share your email with anyone else.
                </FormText>
              </FormGroup>
              <FormGroup
                controlId="formBasicPassword"
              >
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Password
                </FormLabel>
                <FormControl
                  placeholder="Password"
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                className="w-100"
                disabled={false}
                type="submit"
                variant="info"
              >
                 Submit
              </Button>
            </CardBody>
          </AccordionCollapse>
        </Card>
      </ForwardRef>
    </Form>
  </Row>
</Container>
`;
